### DISCUSSION TITLE
How to scan container images in ECR in aws account?

### DISCUSSION THREAD

**Discussion by pritam-yadu** [aquasecurity/trivy#3763] [Q&A]
What's the command to scan container images in aws ecr? I am not finding any documentation for it. Is this supported by trivy? If yes, how to achieve that? Do I need to provide any permissions to trivy or will setting aws credentials as environment variable is sufficient.

**Comment by itaysk** [comment]
https://aquasecurity.github.io/trivy/v0.38/docs/target/container_image/#container-registry
https://aquasecurity.github.io/trivy/v0.38/docs/advanced/private-registries/ecr/

**Comment by pritam-yadu** [comment]
Thanks for reply @itaysk . If I understood correctly, does it mean that trivy first downloads the image locally from container registry and then scans it?

**Reply by knqyf263** [reply]
Yes and no. The image is pulled and processed in memory.

**Comment by munishchouhan** [comment]
how does this works when I am using trivy docker image to scan images from AWS ECR?

**Reply by knqyf263** [reply]
same

**Comment by stevekm** [comment]
I found that the docs listed on this page were not helpful at all for scanning a container that is hosted on your AWS ECR. 

This page was helpful https://stackoverflow.com/questions/70516891/scanning-local-docker-image-for-vulnerability-using-trivy-gives-unauthorized

The solution I found was this;

```bash
# make sure you are logged into your AWS ECR 

aws ecr get-login-password --region "us-east-1" | docker login --username AWS --password-stdin 01234567890.dkr.ecr.us-east-1.amazonaws.com

# pull the ECR container locally first

docker pull 01234567890.dkr.ecr.us-east-1.amazonaws.com/ubuntu:22.04

# extra: set up your local Trivy cache dir
export TRIVY_CACHE=$HOME/.cache/trivy
mkdir -p "$TRIVY_CACHE"

# run Trivy from the Docker container with the docker socket

docker run --userns=host -v /var/run/docker.sock:/var/run/docker.sock -v "$TRIVY_CACHE":"$TRIVY_CACHE" aquasec/trivy image --cache-dir "$TRIVY_CACHE" 01234567890.dkr.ecr.us-east-1.amazonaws.com/ubuntu:22.04
```

This seems to be working for me now.
